app/controllers/schools/add_participants/base_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Schools
+  module AddParticipants
+    class BaseController < ::Schools::BaseController
+      FORM_SESSION_KEY = :add_participant_form
+      FORM_PARAM_KEY = :schools_add_participant_form
+
+      skip_after_action :verify_authorized
+      skip_after_action :verify_policy_scoped
```

Comment:
These shouldn't be skipped (and you're using `set_school_cohort` so the second one doesn't even need to be)

Reply:
I think we can have a longer ping-pong like discussion about those :D IMO, they not only should be skipped, but they should not exist at all. Authorisation is way too complex and important part of the system to be guarded by those terribly hacked methods. As mentioned earlier, they are very easy to be walked around. I prefer no safety net over a safety net that doesn't work.

Reply:
My actual point was that there is no authorization being done in this controller.